central/cve/telemetry/vulnerabilities.go

```diff
@@ -0,0 +1,167 @@
+package telemetry
+
+import (
+	"context"
+	"strconv"
+
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/search"
+)
+
+// Label is an alias because the central/metrics package cannot import it.
+type Label = string
+
+var labelOrder = map[Label]int{
+	"Cluster":          1,
+	"Namespace":        2,
+	"Deployment":       3,
+	"ImageID":          4,
+	"ImageRegistry":    5,
+	"ImageRemote":      6,
+	"ImageTag":         7,
+	"Component":        8,
+	"ComponentVersion": 9,
+	"CVE":              10,
+	"CVSS":             11,
+	"OperatingSystem":  12,
+	"Severity":         13,
+	"SeverityV2":       14,
+	"SeverityV3":       15,
+	"IsFixable":        16,
+}
+
+type record struct {
+	labels map[Label]string
+	total  int
+}
+
+type result map[metricName]map[metricKey]*record
+
+func (h *vulnerabilityMetricsImpl) trackVulnerabilityMetrics(ctx context.Context) result {
+	metrics := make(result)
+	for metric := range h.metrics {
+		metrics[metric] = make(map[metricKey]*record)
+	}
+	// Optimization opportunity:
+	// The resource filter is known at this point, so a more precise query could be constructed here.
+	_ = h.ds.WalkByQuery(ctx, search.EmptyQuery(), func(deployment *storage.Deployment) error {
+		return h.trackDeployment(ctx, metrics, deployment)
+	})
+	return metrics
+}
+
+func (h *vulnerabilityMetricsImpl) trackDeployment(ctx context.Context, aggregated result, deployment *storage.Deployment) error {
```

Comment:
This could cause a spike in memory and db usage since we go through all the images and vulns for all deployments. Since, this job can run every hour, it would be good to measure the load on large clusters.